### ISSUE TITLE
allow array indexing lookup of a value

### ISSUE THREAD

**Issue by jtarchie** [cloudfoundry/bosh-cli#549]
# Request

With `vars.yml`,

```yaml
people: [ Bob, Susan ]
```

With `test.yml`,

```yaml
someone: ((people.1))
```

When I run the command `bosh int test.yaml -l vars.yml`,
it should output the following:

```yaml
someone: Susan
```

# Currently
When the above is run, it produces an error,

```
Finding variable 'people.1':
  Expected to find a map at path '/1' but found '[]interface {}'
```

This is because `bosh int` assumes all path lookups will be hash-maps.

# Investigation

This appears to be an issue with [bosh-cli creation](https://github.com/cloudfoundry/bosh-cli/blob/6e7218cff8a8dffde32942a9800dcd0df8b5fb3d/director/template/template.go#L178) of a `gopatch.FindOp`.

**Comment by jtarchie** [comment]
I have a PR coming, but after investigation it appears that `go-path` will need to be updated, too.
When switching to `patch.NewPointerFromString`, is appears that an index value always assumes a array, not an array or a hash with that index key.

**Comment by jtarchie** [comment]
After talking to @cppforlife, no changes were made to `go-patch` as this is a very specific feature to bosh interpolation.
The PR #550 reflects this change.

**Comment by genevieve** [comment]
@jtarchie It appears a new release of the bosh-cli was cut with your PR: https://github.com/cloudfoundry/bosh-cli/releases/tag/v6.2.0

We can call this out as a dependency in the readme.

**Comment by genevieve** [comment]
Think this issue can be closed though!

**Comment by xtreme-conor-nosal** [comment]
merged in 6.2.0
